src/main/java/com/datalogics/pdf/samples/manipulation/ConvertToPdfA2.java

```diff
@@ -0,0 +1,183 @@
+        final PDFA2ConversionOptions options = PDFA2ConversionOptionsFactory.getConfiguredPdfA2bInstance(pdfDoc);
+
+        try {
+            // Attempt to convert the PDF to the supplied PDF/A2 conformance level
+            if (PDFA2Service.convert(pdfDoc, conformanceLevel, options, handler)) {
+                final PDFSaveOptions saveOpt = PDFSaveFullOptions.newInstance();
+
+                writer = IoUtils.newByteWriter(outputUrl);
+                pdfDoc.save(writer, saveOpt);
+
+                final String successMsg = "\nConverted output written to: " + outputUrl.toString();
+                LOGGER.info(successMsg);
+            } else {
+                LOGGER.info("Errors encountered when converting document.");
+            }
+        } finally {
+            if (writer != null) {
+                writer.close();
+            }
+            if (pdfDoc != null) {
+                pdfDoc.close();
+            }
+        }
+    }
+
+    /**
+     * Validates an input PDF document for PDF/A-2 specification with whatever conformance level was supplied.
+     *
+     * @param inputUrl The URL of the document to be validated
+     * @param conformanceLevel The conformance level of the PDF/A-2 spec the document should conform to
+     * @throws PDFInvalidDocumentException a general problem with the PDF document, which may now be in an invalid state
+     * @throws PDFIOException there was an error reading or writing a PDF file or temporary caches
+     * @throws PDFSecurityException some general security issue occurred during the processing of the request
+     * @throws IOException an I/O operation failed or was interrupted
+     * @throws PDFUnableToCompleteOperationException the operation was unable to be completed
+     * @throws PDFInvalidParameterException one or more of the parameters passed to a method is invalid
+     */
+    public static boolean validate(final URL inputUrl, final PDFA2ConformanceLevel conformanceLevel)
+                    throws PDFInvalidDocumentException, PDFIOException, PDFSecurityException, IOException,
+                    PDFUnableToCompleteOperationException, PDFInvalidParameterException {
+        final PDFDocument doc = DocumentUtils.openPdfDocument(inputUrl);
+        final PdfA2ValidationHandler validationHandler = new PdfA2ValidationHandler();
+        final boolean validity = PDFA2Service.validate(doc, conformanceLevel, new PDFA2ValidationOptions(),
+                                                       validationHandler);
+
+        doc.close();
+
+        if (validity) {
+            LOGGER.info("Validation succeeded.");
+        } else {
+            LOGGER.info("Validation failed");
+        }
+        return validity;
+    }
+
+    private static final class PdfA2ConversionHandler extends PDFA2DefaultConversionHandler {
+        @Override
+        public boolean conversionSummary(final boolean fixesApplied, final boolean unfixableFound) {
+            return !unfixableFound;
```

Comment:
I had to go look up the doc to understand what the return value is doing. Maybe a comment here that says something like `// continue processing unless unfixable errors were found` would have helped me understand faster.